src/cace/amm/typing.c

```diff
@@ -523,6 +523,74 @@ static int builtin_real64_convert(const amm_type_t *self, ari_t *out, const ari_
     }
 }
 
+static bool builtin_time_constraints(struct timespec *out, const ari_t *ari)
+{
+    const ari_lit_t *obj = &(ari->as_lit);
+    struct timespec value;
+    switch (obj->prim_type)
+    {
+        case ARI_PRIM_UINT64:
+            value.tv_sec = obj->value.as_uint64;
+            value.tv_nsec = 0;
+            break;
+        case ARI_PRIM_INT64:
+            value.tv_sec = obj->value.as_int64;
+            value.tv_nsec = 0;
+            break;
+        case ARI_PRIM_FLOAT64:
+        {
+            double v = obj->value.as_float64;
+            if (isnan(v) || isinf(v) || v > INT64_MAX || v < 0.0)
+            {
+              return 1;
+            }
+
+            double integral;
+            double frac = modf(v, &integral);
```

Comment:
Can you include some unit tests (in `test_typing.c`) for this behavior with negative values? It might need a little adjusting but test should confirm. I didn't realize that the time-specific conversion was missing earlier so this is a good catch.

Reply:
Good call! Added test cases and also fixed the return value when invalid floating point numbers (including negatives) are found.

Reply:
That's fair, modified to allow negative values, and adjusted test cases accordingly.